cmake/cmakepp_core/map/update.cmake

```diff
@@ -0,0 +1,36 @@
+include_guard()
+include(cmakepp_core/algorithm/copy)
+include(cmakepp_core/asserts/signature)
+
+#[[[ Adds the key-value pairs from one dictionary to another.
+#
+# This function is to add all the key-value pairs from one dictionary into
```

Comment:
IMHO "update" is a bit generic of a name; how about "union" or "merge"?

Reply:
Also you should specify that it overwrites existing keys in the "this" instance.